common/keydb/cache/keydb.go

```diff
@@ -0,0 +1,69 @@
+package cache
+
+import (
+	"context"
+	"errors"
+
+	"github.com/matrix-org/dendrite/common/caching"
+	"github.com/matrix-org/dendrite/common/keydb"
+	"github.com/matrix-org/gomatrixserverlib"
+)
+
+// A Database implements gomatrixserverlib.KeyDatabase and is used to store
+// the public keys for other matrix servers.
+type Database struct {
```

Comment:
Naming: would prefer `KeyDatabase` so it reads better - `cache.NewKeyDatabase(...)` as `cache.NewDatabase(...)` is vague.